### PR TITLE
Sprint 4

### PR DESCRIPTION
**Pull request by Praveen-Nucleusteq**

### CODE REVIEW

File: Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Controller/EmployeeController.java [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+     * @param updatedDetails  A map containing updated
+     *  details like projectId and managerId.
+     * @return A ResponseDto representing the result of the update operation.
+     */
+    @PutMapping("/updateEmployee/{id}")
+    public final ResponseDto assignProject(@PathVariable final Long id,
+            @RequestBody final  Map<String, Long> updatedDetails) {
+        Long projectId = updatedDetails.get("projectId");
+        Long managerId = updatedDetails.get("managerId");
+        if (projectId == null || managerId == null) {
```
Comment (aayushi-nucleusteq) [inline Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Controller/EmployeeController.java:162]: apply validations at service level or in validation class

File: Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Exception/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            String fieldName = ((FieldError) error).getField();
+            String message = error.getDefaultMessage();
+            resp.put(fieldName, message);
+        });
+        return resp;
+    }
+
+//    /**
+//     * Handle the CustomException exception by returning a
+//      NOT_ACCEPTABLE status and a ResponseDto with the error message.
```
Comment (aayushi-nucleusteq) [inline Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Exception/GlobalExceptionHandler.java:70]: remove commented codes

File: Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Service/AdminService.java [hunk 3]
```diff
@@ -1,100 +1,273 @@
...
+     * Retrieve a list of requested resources as RequestResourceOutDto objects.
+     *
+     * @return A list of RequestResourceOutDto objects
+     * representing the requested
+     *         resources.
+     */
+    public final List<RequestResourceOutDto> requestedResource() {
+        List<RequestResource> requestResourceList = this.
+                requestResourceRepository.findAll();
+        System.out.println(requestResourceList + "praveen");
```
Comment (aayushi-nucleusteq) [inline Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Service/AdminService.java:105]: remove sysouts, use loggers instead. remove commented codes as well

File: Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Service/EmployeeService.java [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+     */
+    @Autowired
+    private ProjectRepository projectRepository;
+
+
+//    /**
+//     * Constructs an EmployeeService with the necessary dependencies.
+//     *
+//     * @param adminsRepository The repository for managing Admin entities.
+//     * @param modelMappers     The ModelMapper for mapping between DTOs and
```
Comment (aayushi-nucleusteq) [inline Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Service/EmployeeService.java:56]: remove commented codes

File: Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Validation/Validation.java [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+     * @param employeeInDto employee id
+     * @throwsFound exception
+     */
+    public final void checkEmployeeIdExists(final
+            EmployeeInDto employeeInDto) {
+        if (employeeRepository.findByEmpId(employeeInDto.
+                getEmpId()) != null) {
+            LOGGER.error("Employee id already exists");
+                throw new UserAlreadyFound(
+                        "Employee id already exists");
```
Comment (aayushi-nucleusteq) [inline Capstone/Backend/EmployeeManagementSystem/src/main/java/com/backend/EMS/Validation/Validation.java:117]: Make a utility class, do not pass string message directly

File: Capstone/Backend/EmployeeManagementSystem/src/test/java/com/backend/EMS/DTO/ProjectOutDtoTest.java [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        teamMembers.add("Jane");
+        dto.setTeam(teamMembers);
+        assertEquals(teamMembers, dto.getTeam());
+    }
+
+    @Test
+    public void testNullValues() {
+        // Test getting null values for all fields
+//        assertNull(dto.getId());
+//        assertNull(dto.getProjectName());
```
Comment (aayushi-nucleusteq) [inline Capstone/Backend/EmployeeManagementSystem/src/test/java/com/backend/EMS/DTO/ProjectOutDtoTest.java:67]: remove commented codes

File: Capstone/Backend/EmployeeManagementSystem/src/test/java/com/backend/EMS/Model/EmployeeTest.java [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+	void testGetId() {
+		  employee.setId(1L);
+	      assertEquals(1L, employee.getId());
+	}
+
+	@Test
+	void testSetId() {
+		  employee.setId(1L);
+	      assertEquals(1L, employee.getId());
+//	      assertThrows(IllegalArgumentException.class, () -> admin.setEmpId(null));
```
Comment (aayushi-nucleusteq) [inline Capstone/Backend/EmployeeManagementSystem/src/test/java/com/backend/EMS/Model/EmployeeTest.java:27]: remove commented codes from everywhere

File: Capstone/Backend/EmployeeManagementSystem/src/test/java/com/backend/EMS/Model/EmployeeTest.java [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        employee.setPassword("myPassword");
+        assertEquals("myPassword", employee.getPassword());
+    }
+
+//    @Test
+//    void testGetConfirmPassword() {
+//        admin.setConfirmPassword("myPassword");
+//        assertEquals("myPassword", admin.getConfirmPassword());
+//    }
+//
```
Comment (aayushi-nucleusteq) [inline Capstone/Backend/EmployeeManagementSystem/src/test/java/com/backend/EMS/Model/EmployeeTest.java:150]: remove commented codes

**Review by aayushi-nucleusteq — CHANGES_REQUESTED** [review]

**Comment by Praveen-Nucleusteq** [comment]
Resolves

**Pull request closed without merging**